spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -401,7 +401,7 @@ private Expression conflictDetectionFilter() {
         filter = Expressions.and(filter, expr);
       }
 
-      return filter;
+      return SERIALIZABLE == isolationLevel ? Expressions.alwaysTrue() : filter;
```

Comment:
If we are going with this approach, it is better to check the isolation level early before line 395 and return.